### PR TITLE
[NzNuXsIR/27] comment

### PR DESCRIPTION
**Pull request by mintrudev**
# What's this PR/MR do?
1. UI:
- Comment
2. Function: N/A

3. Impact range: N/A

# Task link
- https://trello.com/c/NzNuXsIR/27-comments

# TODO Checkboxes
- [x] Did you check UI is as same as Figma Design?
- [x] Did you check Spec/Flow?
- [x] Did you note influence scope + pull request link in sub task?
- [x] Did you check all test cases?
  1. Desktop
  2. Mobile

# Screenshot (UI task)
| Desktop | image |
| --- | --- |
| Mobile |  image

### CODE REVIEW

File: components/Comment/EditorForm/index.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    <div className={cn(`flex gap-2`, className)}>
+      <Avatar>
+        <AvatarImage src="https://github.com/shadcn.png" />
+        <AvatarFallback>CN</AvatarFallback>
+      </Avatar>
+      <form className="flex w-full flex-col gap-3">
+        <MdEditor
+          style={{ height: "160px" }}
+          // onChange={handleEditorChange}
+          renderHTML={(text) => mdParser.render(text)}
```
Comment (tranduybau) [inline components/Comment/EditorForm/index.tsx]:
```suggestion
          renderHTML={mdParser.render}
```

File: components/Comment/EditorForm/index.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+export default function EditorForm(props: Props) {
+  const { dictionary, className } = props;
+
+  return (
+    <div className={cn(`flex gap-2`, className)}>
+      <Avatar>
+        <AvatarImage src="https://github.com/shadcn.png" />
+        <AvatarFallback>CN</AvatarFallback>
+      </Avatar>
+      <form className="flex w-full flex-col gap-3">
```
Comment (tranduybau) [inline components/Comment/EditorForm/index.tsx]:
```suggestion
      <form className="flex w-full flex-col gap-3 [&>*:fist-child]:h-40">
```

File: components/Comment/HeaderComment/index.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import React from "react";
+
+import { cn } from "@/lib/utils";
+import { LocaleKeys } from "@/types/locales";
+
+import { Button } from "../../ui/button";
```
Comment (tranduybau) [inline components/Comment/HeaderComment/index.tsx]:
```suggestion
import { Button } from "@/components/ui/button";
```

File: components/Comment/SingleComment/index.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    className,
+    author,
+    comment,
+    commentTime,
+    like,
+    image,
+    href,
+    dictionary,
+  } = props;
+  return (
```
Comment (tranduybau) [inline components/Comment/SingleComment/index.tsx]:
```suggestion

  return (
```

File: components/Comment/SingleComment/index.tsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+              aria-label="More menu"
+              width={30}
+              height={30}
+            />
+          </button>
+        </header>
+        <p className="mb-4">{comment}</p>
+
+        <footer className="mt-1 flex gap-4">
+          <button
```
Comment (tranduybau) [inline components/Comment/SingleComment/index.tsx]: Sử dụng AppButton

File: components/Comment/index.tsx [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    <div className="container">
+      <HeaderComment
+        dictionary={dictionary}
+        className="mb-6"
+        numberComment={data.length}
+      />
+
+      <EditorForm dictionary={dictionary} className="mb-6" />
+      {data.map((comment) => {
+        return (
```
Comment (tranduybau) [inline components/Comment/index.tsx]:
```suggestion
      {data.map((comment) => (
```

Comment (mintrudev) [reply components/Comment/index.tsx]: đã fix

File: components/Comment/EditorForm/index.tsx [hunk 7]
```diff
@@ -0,0 +1,49 @@
+"use client";
```
Comment (tranduybau) [inline components/Comment/EditorForm/index.tsx]: `components/comments/editor-form/index.tsx`

File: components/Comment/HeaderComment/index.tsx [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+  return (
+    <div className={cn("flex items-center justify-between", className)}>
+      <h2 className="text-heading-4 font-bold">
+        {dictionary["Top comments"]} ({numberComment})
+      </h2>
+      <Button variant="outline">{dictionary.Subscribe}</Button>
+    </div>
+  );
+}
+export default React.memo(HeaderComment);
```
Comment (tranduybau) [inline components/Comment/HeaderComment/index.tsx]: Fast-compare

File: components/Comment/SingleComment/index.tsx [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+  return (
+    <div className={cn("flex items-start gap-2", className)}>
+      <AppLink href={href} aria-label={author}>
+        <Avatar className="mt-3 h-8 w-8">
+          <AvatarImage src={image} />
+          <AvatarFallback>{author}</AvatarFallback>
+        </Avatar>
+      </AppLink>
+      <article className="w-full rounded-md border border-[#1717170d] px-3">
+        <header className="flex items-center justify-between gap-1 pt-2 text-gray-500">
```
Comment (tranduybau) [inline components/Comment/SingleComment/index.tsx]: Dùng thẻ khác

File: components/Comment/SingleComment/index.tsx [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+          <Button className="flex items-center gap-1 rounded-md bg-white p-1 text-[#3d3d3d] hover:bg-gray-100">
+            <AppIcon
+              src="/svg/reply.svg#id"
+              aria-label="Reply"
+              width={20}
+              height={20}
+            />
+            <span> {dictionary.Reply}</span>
+          </Button>
+        </footer>
```
Comment (tranduybau) [inline components/Comment/SingleComment/index.tsx]: Sửa sang thẻ khác

**Review by tranduybau — APPROVED** [review]

**Pull request merged**